app/src/main/java/ch/epfl/skysync/components/forms/FlightForm.kt

```diff
@@ -98,11 +98,15 @@ fun FlightForm(
         var flightTypeValue: FlightType? by remember { mutableStateOf(currentFlight?.flightType) }
         var flightTypeValueError by remember { mutableStateOf(false) }
 
-        val vehicle: Vehicle? by remember { mutableStateOf(null) }
-        val listVehiclesValue = remember {
-          mutableStateListOf(*currentFlight?.vehicles?.toTypedArray() ?: emptyArray())
+        val chosenVehicles: MutableState<List<Vehicle?>> = remember {
+          if (currentFlight?.vehicles == null) {
+            mutableStateOf(listOf(null))
+          } else if (currentFlight.vehicles.isNotEmpty()) {
+            mutableStateOf(currentFlight.vehicles)
+          } else {
+            mutableStateOf(listOf(null))
+          }
```

Comment:
Is it necessary to have a list of "Vehicle?" instead of a list of "Vehicle" ? And you could also eventually merge conditions as two of them are doing exactly the same thing.